Weather/ViewModels/CityViewModel.swift

```diff
@@ -0,0 +1,204 @@
+//
+//  CityViewModel.swift
+//  Weather
+//
+//  Created by Alex on 31.07.22.
+//
+
+import Foundation
+import SwiftUI
+import CoreLocation
+
+class CityViewModel: ObservableObject {
+    
+    @Published var weather = WeatherResponse.emptyResponse()
+    @Published var city: String = "Minsk" {
+        didSet {
+            getLocation()
+        }
+    }
+    
+    private lazy var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .full
+        return formatter
+    }()
+    
+    private lazy var dayFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "EEE"
+        return formatter
+    }()
+    
+    private lazy var timeFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "hh a"
+        return formatter
+    }()
+    
+    init() {
+        getLocation()
+    }
+    
+    var date: String {
+        return dateFormatter.string(from: Date(timeIntervalSince1970: TimeInterval(weather.current.dt)))
+    }
+    
+    var weatherIcon: String {
+        if weather.current.weather.count > 0 {
+            return weather.current.weather[0].icon
+        }
+        return "sun.max.fill"
```

Comment:
The same here: `weather.current.weather.first?.icon ?? "sun.max.fill"`

Reply:
Fixed.)